cache.rst

```diff
@@ -182,6 +138,71 @@ will create pools with service IDs that follow the pattern ``cache.[type]``.
             ],
         ]);
 
+System Cache and Application Cache
+----------------------------------
+
+There are two pools that are always enabled by default. They are
+``cache.system`` and ``cache.app``.
+
+The system cache ``cache.system`` is used for things like annotations,
+serializer, and validation. Usage is reserved for Symfony components
+and you should not use it for application cache needs.
```

Comment:
The application can also use it if it has its own data derived from the source code that can be generated during cache-warmup.

The rules for using `cache.system` are:
- the cached content does not need to change if the source code of the app does not change (i.e. it does not need to change between deployments)
- the cached content has a CacheWarmer generating it (the `cache.adapter.system` is optimized for being read-only after the warm-up)

That's why it is used for the metadata of serializer and validation. But we could totally imagine applications that generate some metadata extracted from their source code (even though this is indeed more advanced use cases than the ones using `cache.app`.

On the other hand, `cache.app` is independent from the deployment cycle of the source code (and generally needs to use a shared storage like Redis rather than the filesystem once you have multiple servers)

Reply:
@stof Thanks for commenting, I added your feedback to the PR.